yak_ros/src/yak_ros1_node.cpp

```diff
@@ -62,43 +70,63 @@ class OnlineFusionServer
    */
   void onReceivedDepthImg(const sensor_msgs::ImageConstPtr& image_in)
   {
-    // Get the camera pose in the world frame at the time when the depth image was generated.
-    ROS_INFO_STREAM("Got depth image");
-    geometry_msgs::TransformStamped transform_world_to_camera;
-    try
-    {
-      transform_world_to_camera = tf_buffer_.lookupTransform("base_link", image_in->header.frame_id, image_in->header.stamp);
-    }
-    catch(tf2::TransformException &ex)
-    {
-      // Abort integration if tf lookup failed
-      ROS_WARN("%s", ex.what());
-      return;
+    image_q_.push(image_in);
+    if(image_q_.size() > 2){
+      auto next_image = image_q_.front();
+      image_q_.pop(); // remove it now
+
+      // Get the camera pose in the tsdf frame at the time when the depth image was generated.
+      ROS_INFO_STREAM("Got depth image");
+      geometry_msgs::TransformStamped transform_tsdf_to_camera;
+      try
+	{
+	  transform_tsdf_to_camera = tf_buffer_.lookupTransform(tsdf_frame_name_, next_image->header.frame_id, next_image->header.stamp);
+	}
+      catch(tf2::TransformException &ex)
+	{
+	  // Abort integration if tf lookup failed
+	  ROS_WARN("%s", ex.what());
+	  return;
+	}
+      Eigen::Affine3d tsdf_to_camera = tf2::transformToEigen(transform_tsdf_to_camera);
+      
+      // Find how much the camera moved since the last depth image. If the magnitude of motion was below some threshold, abort integration.
+      // This is to prevent noise from accumulating in the isosurface due to numerous observations from the same pose.
+      std::double_t motion_mag = (tsdf_to_camera.inverse() *tsdf_to_camera_prev_).translation().norm();
+      ROS_INFO_STREAM(motion_mag);
+      if(motion_mag < DEFAULT_MINIMUM_TRANSLATION)
+	{
+	  ROS_INFO_STREAM("Camera motion below threshold");
+	  return;
+	}
+      
+      cv_bridge::CvImagePtr cv_ptr = cv_bridge::toCvCopy(next_image, sensor_msgs::image_encodings::TYPE_16UC1);
+
+#if (0)// debug the transform by printing it out every 30'th update
+      static int q=0;
```

Comment:
@DavidMerzJr right now, this only accepts the raw depth image which is a 16UC1. The rectified image is 32FC1 so it does not work. We need the rectified image for intrinsic calibration to have any affect.